addons/mass_mailing/controllers/main.py

```diff
@@ -128,6 +128,9 @@ def view(self, mailing_id, email=None, res_id=None, token=""):
                 'class="o_snippet_view_in_browser" style="display: none;"'
             )
 
+            if res_id:
+                res[mailing_id] = request.env['mail.render.mixin']._render_template(res[mailing_id], mailing.mailing_model_real, [res_id], post_process=True)[res_id]
```

Comment:
This will fail for most people because they will not have the access rights required to render the current values. (tried on your runbot, confirmed even just ${object.email} gives a traceback)

We could use sudo, but then that means allowing people to get updated information just because they received an email in the past, with past information.

Not sure how much of a concern that is so... Calling upon @tde-banana-odoo

Reply:
Thanks for the check!
My bad, sorry. I did not make the test being logged out :man_facepalming:
> We could use sudo, but then that means allowing people to get updated information just because they received an email in the past, with past information.

Indeed, mhhh... I will put back the PR in draft to compare behavior in this specific case ("updated information") with 16.0 :wink:

Reply:
Hello @thiryr , sorry for the delay (was on sick leave)
I just compared with 16.0; I forget to use a sudo() (by using `mailing` variable instead of `request.env['mail.render.mixin']`).
However, anyone that have the 'online view' link will have access to the mail content (same on 16.0) therefore, not sure at the end which is the right behavior...
@tde-banana-odoo will finally need your insight :pray:  (thanks in advance!)